### PR TITLE
Multiply well trans due to perm reduction in salt prec

### PR DESCRIPTION
**Pull request by daavid00**
Requires https://github.com/OPM/opm-simulators/pull/4919

### CODE REVIEW

**Comment by GitPaean** [comment]
jenkins build this opm-simulators=4919 please

File: opm/models/blackoil/blackoilbrinemodules.hh [hunk 1]
```diff
@@ -431,11 +431,13 @@ public:
             if (priVars.primaryVarsMeaningBrine() == PrimaryVariables::BrineMeaning::Sp) {
                 saltSaturation_ = priVars.makeEvaluation(saltConcentrationIdx, timeIdx);
                 fs.setSaltConcentration(saltSolubility_);
+                fs.setSaltSaturation(saltSaturation_);
             }
             else {
                 saltConcentration_ = priVars.makeEvaluation(saltConcentrationIdx, timeIdx);
                 fs.setSaltConcentration(saltConcentration_);
                 saltSaturation_ = 0.0;
+                fs.setSaltSaturation(saltSaturation_);
```
Comment (bska) [inline opm/models/blackoil/blackoilbrinemodules.hh:440]: These two branches both call `fs.setSaltSaturation(saltSaturation_)`, the only difference is the value of `saltSaturation`, so I'd prefer if the function call was moved below the conditional.

**Review by bska — COMMENTED** [review]
I think we should extract common code out of the conditionals here.

File: opm/models/blackoil/blackoilbrinemodules.hh (hunk 1 above)
Comment (daavid00) [reply opm/models/blackoil/blackoilbrinemodules.hh:440]: I thought about it right after I made the pull request, thanks for mentioning this. I will update this after Jenkins is finished.

**Comment by GitPaean** [comment]
jenkins build this opm-simulators=4919 please

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
jenkins build this opm-simulators=4919 please

**Comment by totto82** [comment]
jenkins build this opm-simulators=4919 please

**Comment by totto82** [comment]
The test failures are expected as the PR modifies the well code. @plgbrt or @goncalvesmachadoc Any comments? Do you agree with the change?

**Comment by goncalvesmachadoc** [comment]
The changes in both PR's looks sensible to me.

**Comment by totto82** [comment]
jenkins build this opm-simulators=4919 please

**Comment by totto82** [comment]
jenkins build this opm-simulators=4919 update_data please

**Comment by totto82** [comment]
jenkins build this opm-simulators=4919 update_data please

**Comment by totto82** [comment]
jenkins build this opm-simulators=4919 opm-tests=1079 please

**Comment by totto82** [comment]
Thanks for the fix. Will merge.

**Pull request merged**